File: app/src/main/java/co/cubeflow/parakeet/auth/authenticator/ParakeetAuthenticatorService.java
package co.cubeflow.parakeet.auth.authenticator;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import androidx.annotation.Nullable;

/**
 * @version 1.0.0
 * @since v1.0.0 (11/24/2017)
 */
public class ParakeetAuthenticatorService extends Service {

    private ParakeetAuthenticator mAuthenticator;

    @Override
    public void onCreate() {
        super.onCreate();
        mAuthenticator = new ParakeetAuthenticator(this);
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return mAuthenticator.getIBinder();
    }
}

File: app/src/main/java/co/cubeflow/parakeet/auth/authenticator/ParakeetAuthenticatorService.kt
package co.cubeflow.parakeet.auth.authenticator

import android.app.Service
import android.content.Intent
import android.os.IBinder
import androidx.annotation.Nullable

/**
 * @version 1.0.0
 * @since v1.0.0 (11/24/2017)
 */
class ParakeetAuthenticatorService : Service() {

    private var mAuthenticator: ParakeetAuthenticator? = null

    override fun onCreate() {
        super.onCreate()
        mAuthenticator = ParakeetAuthenticator(this)
    }

    @Nullable
    override fun onBind(intent: Intent): IBinder {
        return mAuthenticator!!.iBinder
    }
}
